cli/src/main.rs

```diff
@@ -43,122 +144,87 @@ impl<T, E: Error> PrettyResult for Result<T, E> {
         match self {
             Result::Ok(value) => value,
             Result::Err(error) => {
-                print_err(error);
+                print_err(&error);
                 std::process::exit(1);
             }
         }
     }
 }
 
 fn main() {
+    if let Err(e) = run() {
+        print_err(e.as_ref());
+        std::process::exit(1);
+    }
+}
+
+/// Error type for the CLI
+#[derive(Debug, Clone)]
+struct CliError(String);
```

Comment:
Would `type CliError = Cow<'static, str>` allow us to write less boilerplate?
Actually, maybe even `type CliError = &'static str`?